packages/sanity/src/_internal/manifest/Icon.tsx

```diff
@@ -0,0 +1,104 @@
+import {COLOR_HUES, hues} from '@sanity/color'
+import {ThemeProvider} from '@sanity/ui'
+import {buildTheme} from '@sanity/ui/theme'
+import {darken, hasBadContrast, lighten, readableColor} from 'color2k'
+import {type ComponentType, createElement, isValidElement, type ReactNode} from 'react'
+import {isValidElementType} from 'react-is'
+import {ServerStyleSheet, styled, StyleSheetManager} from 'styled-components'
+
+function pseudoRandomNumber(seed: string) {
+  const hashCode = seed
+    .split('')
+    // eslint-disable-next-line no-bitwise
+    .reduce((prevHash, currVal) => ((prevHash << 5) - prevHash + currVal.charCodeAt(0)) | 0, 0)
+  return Math.abs((hashCode * 16807) % 2147483647) / 2147483647
+}
+
+const SvgText = styled.text`
+  font-family: ${({theme}) => theme.sanity.fonts.text.family};
+  font-weight: ${({theme}) => theme.sanity.fonts.text.weights.medium};
+  font-size: ${({theme}) => theme.sanity.fonts.text.sizes[1].fontSize}px;
+  transform: translateY(1px);
+`
+
+/**
+ * Creates an icon element based on the input title
+ */
+function createDefaultIcon(title: string, subtitle: string): JSX.Element {
+  const rng1 = pseudoRandomNumber(`${title} ${subtitle}`)
+
+  const huesWithoutGray = COLOR_HUES.filter((hue) => hue !== 'gray')
+  const colorHue = huesWithoutGray[Math.floor(rng1 * huesWithoutGray.length)]
+  const possibleTints = ['300', '400', '500', '600', '700'] as const
+  const rng2 = pseudoRandomNumber(rng1.toString())
+  const tint = possibleTints[Math.floor(rng2 * possibleTints.length)]
+  const color = hues[colorHue][tint].hex
+
+  const letters = title
+    // split by whitespace
+    .split(/\s/g)
+    // replace all non-word characters with empty string
+    .map((word) => word.replace(/\\W/g, ''))
+    // remove empty strings
+    .filter(Boolean)
+    // take the first two words
+    .slice(0, 2)
+    // grab the first letter and make it upper case
+    .map((i) => i.charAt(0).toUpperCase())
+
+  const darkened = darken(color, 0.4)
+  const lightened = lighten(color, 0.4)
+
+  /* eslint-disable no-negated-condition */
```

Comment:
I can do, I copied from https://github.com/sanity-io/sanity/blob/feat/manifest-additions/packages/sanity/src/core/config/createDefaultIcon.tsx because it wouldn't let me import it here due to "boundary crossing" is there an easier way to share code? I don't mind tweaking it & adding tests etc.

Reply:
I'll defer to Studio team on what to do here: Seems to me that it should be extracted and reused both places.

Reply:
i'd prefer that, lets hold fire till they respond 🫡

Reply:
You can export it from the main `sanity` export and mark it as internal/hidden